### PR TITLE
Feature/refactor estimation years

### PR DESCRIPTION
**Pull request by collijk**
This handles half of the estimation year refactor (the raw validator part).  Sim validator section coming next.

Also addressed leftover constants from the last pr.

### CODE REVIEW

File: src/vivarium_inputs/validation/utilities.py [hunk 1]
```diff
@@ -11,23 +11,17 @@
...
 
 
-def check_years(data: pd.DataFrame, year_type: str, error: bool = True):
+def check_years(data: pd.DataFrame, year_type: str, estimation_years: pd.Series) -> None:
     """Check that years in passed data match expected range based on type.
 
     Parameters
     ----------
-    data
+    data :
```
Comment (kfwilson) [inline src/vivarium_inputs/validation/utilities.py]: I don't think you do the : if you're not specifying type - at least, that's what the numpy docstring guide I've been following says (https://numpydoc.readthedocs.io/en/latest/format.html). Let me know if we want to always specify the colon though and I can update the doc strings I'm writing

File: src/vivarium_inputs/validation/utilities.py [hunk 2]
```diff
@@ -11,23 +11,17 @@
...
         Dataframe containing 'year_id' column.
-    year_type
+    year_type :
         String 'annual' or 'binned' indicating expected year range.
```
Comment (kfwilson) [inline src/vivarium_inputs/validation/utilities.py]: This should also include 'either' now right?

File: src/vivarium_inputs/validation/utilities.py (hunk 1 above)
Comment (collijk) [reply src/vivarium_inputs/validation/utilities.py]: Ah, no, we should do the numpy thing.  I've been putting colons in all our docs forever though :-1:

File: src/vivarium_inputs/validation/utilities.py (hunk 2 above)
Comment (collijk) [reply src/vivarium_inputs/validation/utilities.py]: yep

**Pull request merged**